src/generic/stage2_build_tape.h

```diff
@@ -428,29 +529,36 @@ unified_machine(const uint8_t *buf, size_t len, ParsedJson &pj) {
     /* we found an array inside an array, so we need to increment the depth
      */
     depth++;
-    if (depth >= pj.depth_capacity) {
-      goto fail;
+    if (depth < pj.depth_capacity) {
+      goto array_begin;
     }
-    goto array_begin;
+    goto fail; 
+    
   }
   default:
     goto fail;
   }
 
 array_continue:
   UPDATE_CHAR();
-  switch (c) {
-  case ',':
+  // Next bit could be a switch case. Short switch cases tend to be compiled
+  // as sequences of branches. The compiler can't tell which branch is more likely
+  // but we *know* that most arrays contain more than one entry so that ',' is a common result,
+  // hence we build our own sequence of branches, starting with the most likely result.
```

Comment:
This makes sense to me, but I'm curious: did this help?

Reply:
I don't expect that it did, but I am trying to leave nothing to chance.

Reply:
OK. I guess I don't mind the code being one way or the other, but can you please include a note that we're not sure whether this helps, so that when I or someone comes in there we know it *might* be changeable to a switch if we needed?

Reply:
Yes. Will do so right now. Note that with a switch-case (the current setup in master), you are leaving it up to the compiler but it is unclear how the compiler would know (at compile time) what the right choice is.